### PR TITLE
nux: init at 0.1.4

### PR DESCRIPTION
**Pull request by jakobneufeld**
###### Motivation for this change

I want to add my package `nux`.
This is a totally new package!
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jakobneufeld** [comment]
This is my first new nix package, so I probably messed something up!

**Comment by jakobneufeld** [comment]
I fixed the whitespace!

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -30394,7 +30394,7 @@ with pkgs;
   neverball = callPackage ../games/neverball { };
 
   nexuiz = callPackage ../games/nexuiz { };
-
+  nux = callPackage ../tools/misc/nux { };
```
Comment (jakobneufeld) [inline pkgs/top-level/all-packages.nix:30397]: I fixed whitespace here.

File: pkgs/tools/misc/nux/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    installManPage $releaseDir/build/nux-*/out/nux.1
+    installShellCompletion $releaseDir/build/nux-*/out/nux.{bash,fish}
+    installShellCompletion $releaseDir/build/nux-*/out/_nux
+
+  '';
+  nativeBuildInputs = [ asciidoctor installShellFiles ];
+  buildInputs = [pkg-config];
+
+  meta = with lib; {
+    description = "A wrapper over the nix cli.";
```
Comment (legendofmiracles) [inline pkgs/tools/misc/nux/default.nix:32]:
```suggestion
    description = "A wrapper over the nix cli";
```

File: pkgs/tools/misc/nux/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    homepage = "https://github.com/NuxPackage/nux";
+    license = with licenses; [ gpl3 ];
+    mainProgram = "nux";
```
Comment (legendofmiracles) [inline pkgs/tools/misc/nux/default.nix:35]:
```suggestion
```
This is default.

File: pkgs/tools/misc/nux/default.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  cargoSha256 = "sha256-wfUr3dcdALMEgJ6CaXhK4Gqk6xflCnov9tELA63drV4=";
+
+  preFixup = ''
+    installManPage $releaseDir/build/nux-*/out/nux.1
+    installShellCompletion $releaseDir/build/nux-*/out/nux.{bash,fish}
+    installShellCompletion $releaseDir/build/nux-*/out/_nux
+
+  '';
+  nativeBuildInputs = [ asciidoctor installShellFiles ];
+  buildInputs = [pkg-config];
```
Comment (legendofmiracles) [inline pkgs/tools/misc/nux/default.nix:29]:
```suggestion
  buildInputs = [ pkg-config ];
```

File: pkgs/tools/misc/nux/default.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+  meta = with lib; {
```
Comment (legendofmiracles) [inline pkgs/tools/misc/nux/default.nix:31]: You need to add yourself as a maintainer. You can do this by editing maintainers/maintainers.nix in a separate commit: `maintainers: add <name>`

**Review by legendofmiracles — COMMENTED** [review]
Thank you for your contribution! I left some nit-pick comments, but all in all I'm very impressed with the quality of your first PR

**Comment by jakobneufeld** [comment]
Thanks!

**Comment by jakobneufeld** [comment]
Is there anything else I should do @legendofmiracles ?

File: maintainers/maintainer-list.nix [hunk 6]
```diff
@@ -2617,6 +2617,12 @@
     githubId = 841969;
     name = "David Kuo";
   };
+  drzoidberg = {
+    email = "jakob@mast3rsoft.com";
+    github = "jakobneufeld";
+    githubId = "24791219"
```
Comment (r-burns) [inline maintainers/maintainer-list.nix:2623]:
```suggestion
    githubId = "24791219";
```

fixes CI error

File: maintainers/maintainer-list.nix [hunk 7]
```diff
@@ -2617,6 +2617,12 @@
     githubId = 841969;
     name = "David Kuo";
   };
+  drzoidberg = {
```
Comment (r-burns) [inline maintainers/maintainer-list.nix:2620]: could you please sort your entry alphabetically?

Comment (jakobneufeld) [reply maintainers/maintainer-list.nix:2620]: I will.

**Comment by jakobneufeld** [comment]
Better @r-burns?

File: maintainers/maintainer-list.nix [hunk 8]
```diff
@@ -2617,7 +2617,7 @@
     githubId = 841969;
     name = "David Kuo";
   };
-  davidak = {
+   davidak = {
```
Comment (r-burns) [inline maintainers/maintainer-list.nix:2620]:
```suggestion
  davidak = {
```

File: maintainers/maintainer-list.nix [hunk 9]
```diff
@@ -3073,6 +3073,13 @@
     githubId = 10198051;
     name = "Drew Risinger";
   };
+   drzoidberg = {
+    email = "jakob@mast3rsoft.com";
+    github = "jakobneufeld";
+    githubId = "24791219";
+    name = "Jakob Neufeld";
+  };
+
```
Comment (r-burns) [inline maintainers/maintainer-list.nix:3082]:
```suggestion
```

**Review by r-burns — COMMENTED** [review]
Close, I think you picked up a couple extra whitespace changes though

**Comment by jakobneufeld** [comment]
This should be correct @r-burns.

Also thanks for reviewing so quickly, because I am completely new to nixpkgs!

File: pkgs/tools/misc/nux/default.nix [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+
+  '';
+  nativeBuildInputs = [ asciidoctor installShellFiles ];
+  buildInputs = [ pkg-config ];
+
+  meta = with lib; {
+    description = "A wrapper over the nix cli";
+    homepage = "https://github.com/NuxPackage/nux";
+    license = with licenses; [ gpl3 ];
+    maintainers = [ drzoidberg ];
```
Comment (r-burns) [inline pkgs/tools/misc/nux/default.nix:35]:
```suggestion
    maintainers = with maintainers; [ drzoidberg ];
```

Hopefully this will make the CI happy!

**Comment by jakobneufeld** [comment]
Sorry again, can you check @r-burns.

**Comment by jakobneufeld** [comment]
@r-burns, do I have to do something?

**Comment by legendofmiracles** [comment]
All package related commits should be squashed into `nux: init at 0.1.4`

**Comment by jakobneufeld** [comment]
What do you mean @legendofmiracles?

**Comment by legendofmiracles** [comment]
Squash all package commits into one: https://gitbetter.substack.com/p/how-to-squash-git-commits

And all maintainers.nix commits into one as well.

**Comment by jakobneufeld** [comment]
Ok, I will squash these commits. Do I make a new PR @legendofmiracles

**Comment by r-burns** [comment]
No need to make a new PR, you can squash and force push here, thanks!

**Comment by jakobneufeld** [comment]
Sorry, I messed up the rebase!
I will retry!

**Comment by jakobneufeld** [comment]
I will make another fork with all the changes in one @legendofmiracles @r-burns.

**Comment by jakobneufeld** [comment]
@legendofmiracles @r-burns, here is my new PR https://github.com/NixOS/nixpkgs/pull/147021

**Comment by legendofmiracles** [comment]
You could have also force pushed with the -f flag: `git push origin master -f`

**Comment by jakobneufeld** [comment]
@r-burns @legendofmiracles ok, should I  reopen the pr. I am confused The problem is I have other comits squashed and I am confused

**Comment by jonringer** [comment]
You probably want to clean up the git history.

I have a video demonstrating `git rebase -i`

https://www.youtube.com/watch?v=OR8Q-4PLalc

**Comment by legendofmiracles** [comment]
Ah no, the new PR is fine. 
Except for one thing, the maintainers commit shouldn't have 
been squashed into the package commit. 
After that it looks good to me

**Comment by jakobneufeld** [comment]
OK @legendofmiracles. I will make a seperate commit for maintainers. @jonringer thanks for the video!

**Comment by jakobneufeld** [comment]
I fixed everything on the new PR  #147021
please continue reviewing on the new pr @legendofmiracles @r-burns. Thanks so much for being so patient!

**Pull request closed without merging**